### PR TITLE
8328158: Convert java/awt/Choice/NonFocusablePopupMenuTest to automatic main test

### PR DESCRIPTION
**Pull request by azuev-java**
Convert test to the automatic main and move it from its own folder since it is no longer required.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8328158](https://bugs.openjdk.org/browse/JDK-8328158): Convert java/awt/Choice/NonFocusablePopupMenuTest to automatic main test (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18300/head:pull/18300` \
`$ git checkout pull/18300`

Update a local copy of the PR: \
`$ git checkout pull/18300` \
`$ git pull https://git.openjdk.org/jdk.git pull/18300/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18300`

View PR using the GUI difftool: \
`$ git pr show -t 18300`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18300.diff">https://git.openjdk.org/jdk/pull/18300.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18300#issuecomment-1997347459)

### CODE REVIEW

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            choice.add("444");
+            choice.setFocusable(false);
+            this.add(choice);
+            this.setLayout (new FlowLayout());
+            setSize (200,200);
+            setLocationRelativeTo(null);
+            setVisible(true);
+        });
+        robot.waitForIdle();
+        SwingUtilities.invokeAndWait(() -> {
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:65]: Is it ok to have SwingUtilities.invokeAndWait? 
Or should we use EventQueue.invokeAndWait ?

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            pos = choice.getLocationOnScreen();
+            size = choice.getSize();
+        });
+        int selected = choice.getSelectedIndex();
+        robot.mouseMove(pos.x + size.width / 2, pos.y + size.height / 2);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        robot.delay(500);
```
[the remark this answers is not in the document]
Comment (azuev-java) [reply test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:74]: Giving native peer some time to process request to populate a popup from choice. 500 ms seems reasonable.

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            setSize (200,200);
+            setLocationRelativeTo(null);
+            setVisible(true);
+        });
+        robot.waitForIdle();
+        SwingUtilities.invokeAndWait(() -> {
+            pos = choice.getLocationOnScreen();
+            size = choice.getSize();
+        });
+        int selected = choice.getSelectedIndex();
```
[the remark this answers is not in the document]
Comment (azuev-java) [reply test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:69]: Not sure it poses any significant problem but i will move it to EDT.

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        robot.mouseMove(pos.x + size.width / 2, pos.y + size.height * 2);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        robot.delay(500);
+        SwingUtilities.invokeAndWait(() -> {
+            setVisible(false);
+            dispose();
+        });
+        if (choice.getSelectedIndex() == selected) {
```
[the remark this answers is not in the document]
Comment (azuev-java) [reply test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java]: Fixed.

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        robot.setAutoDelay(100);
+        EventQueue.invokeAndWait(() -> {
+            choice = new Choice();
+            choice.add("111");
+            choice.add("222");
+            choice.add("333");
+            choice.add("444");
+            choice.setFocusable(false);
+            this.add(choice);
+            this.setLayout (new FlowLayout());
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:59]:
```suggestion
            this.setLayout(new FlowLayout());
```

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+            setSize (200,200);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:60]:
```suggestion
            setSize (200, 200);
```

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+            setLocationRelativeTo(null);
+            setVisible(true);
+        });
+        robot.waitForIdle();
+        EventQueue.invokeAndWait(() -> {
+            pos = choice.getLocationOnScreen();
+            size = choice.getSize();
+        });
+        EventQueue.invokeAndWait(() -> {
+            selection1 = choice.getSelectedIndex();
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:70]: May be moved inside the EventQueue above.

File: test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        robot.delay(500);
+        robot.mouseMove(pos.x + size.width / 2, pos.y + size.height * 2);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+        robot.waitForIdle();
+        EventQueue.invokeAndWait(() -> {
+            selection2 = choice.getSelectedIndex();
+        });
+        EventQueue.invokeAndWait(() -> {
+            setVisible(false);
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:85]: I think this also can be moved inside above EventQueue.

Comment (azuev-java) [reply test/jdk/java/awt/Choice/NonFocusablePopupMenuTest.java:85]: Sure.

**Review by kumarabhi006 — APPROVED** [review]
LGTM.

**Review by honkar-jdk — APPROVED** [review]
LGTM

**Comment by DamonFool** [comment]
Hi, this test should be marked as headful.
Please see: https://github.com/openjdk/jdk/pull/18326
Thanks.

**Pull request closed without merging**